### PR TITLE
fix(collapsible-panel): Accessible panel Header element

### PR DESCRIPTION
**Pull request by jonnybel**
#### Summary

This PR addresses this problem: https://github.com/commercetools/merchant-center-frontend/pull/8565#issuecomment-598637568.

It occurs when the `CollapsiblePanel` is passed any child elements for the Header that are buttons, because the Header of the CollapsiblePanel is also a `button` element.

#### Description

* Fix the `AccessibleButton` component to become truly accessible when the `as` prop is used:
  * Add `role` and `tabIndex` attributes.
  * Add `onKeyPress` event for `Enter` and `Space` to trigger the `onClick`, for full keyboard navigation support.
* Use this adjusted `AccessibleButton` as a `div` for the CollapsiblePanel header, instead of a `button` element but keeping the a11y.

### CODE REVIEW

File: packages/components/buttons/accessible-button/src/accessible-button.js [hunk 1]
```diff
@@ -9,12 +9,16 @@ import getNormalizedButtonStyles from './accessible-button.styles';
 
 const propsToOmit = ['onClick'];
 
+const getIsEnterOrSpace = e =>
+  e.key === ' ' || e.key === 'Spacebar' || e.which === 13 || e.which === 32;
```
Comment (tdeekens) [inline packages/components/buttons/accessible-button/src/accessible-button.js]: What is `' '`? :). Is there maybe like a itty-bity util library we can use for this? Or do you think it's fine inlining it?

File: packages/components/buttons/accessible-button/src/accessible-button.js [hunk 2]
```diff
@@ -39,9 +45,18 @@ const AccessibleButton = React.forwardRef((props, ref) => {
...
 
-  const buttonProps = {
-    type: props.type,
-  };
+  let buttonProps = {};
+  if (isButton) {
+    buttonProps = {
+      type: props.type,
+    };
+  } else if (!props.isDisabled) {
```
Comment (tdeekens) [inline packages/components/buttons/accessible-button/src/accessible-button.js]: Nit: do-expression?

Comment (jonnybel) [reply packages/components/buttons/accessible-button/src/accessible-button.js]: It gets a little weird-looking with the object braces for the logic I am trying to achieve, am i doing this right?

```
const buttonProps = 
do {
   if (isButton) {
     {
      type: props.type,
    }
   }
  else if (!props.isDisabled) {
    {
      role: 'button',
      tabIndex: '0',
      onKeyPress: event => getIsEnterOrSpace(event) && handleClick(event),
    }
  }
   else {
  { }
 }
```

File: packages/components/buttons/accessible-button/src/accessible-button.js (hunk 1 above)
Comment (jonnybel) [reply packages/components/buttons/accessible-button/src/accessible-button.js]: `' '` is for the Spacebar.

I just double-checked a bit deeper and simplified this line, according to this spec that React uses for the `key` field of their synthetic events: https://www.w3.org/TR/uievents-key/#keys-whitespace

File: packages/components/buttons/accessible-button/src/accessible-button.js (hunk 2 above)
Comment (tdeekens) [reply packages/components/buttons/accessible-button/src/accessible-button.js]: True that also don't make it much simpler either.

Comment (tdeekens) [reply packages/components/buttons/accessible-button/src/accessible-button.js]: ...just tried to avoid the let in a way.

**Pull request merged**